src/helpers/ApiServiceHelper.js

```diff
@@ -29,75 +30,275 @@ function ApiServiceHelper (url = '', clientId = '', clientSecret = '', scope = '
+          .then(response => {
+            if (response.data && response.data.data) {
+              item['patronInfo'] = response.data.data;
+              callback(null, item);
+            } else {
+              callback({
+                holdRequestId: item.id,
+                status: 404,
+                type: 'missing-response-data',
+                message: 'unable to assign the patronInfo key to record. Did not obtain a response.data.data value.'
+              });
+            }
+          })
+          .catch((error) => {
+            // TODO: LOG ERROR
+            callback(Object.assign({}, error, { holdRequestId: item.id }));
+          });
+        } else {
+          callback(
+            HoldRequestConsumerError({
+              message: 'could not generate a patron barcode API url; undefined or empty patron value',
+              type: 'missing-patron-key-value',
+              status: 404,
+              holdRequestId: item.id,
+              error : {
+                data: item,
+              }
+            })
+          );
+        }
+      }, (err, results) => {
+        // Handle the appropriate error for each result
+        if (err) {
+          let errorResponse = Object.assign({}, {
+            type: 'patron-service-api',
+            function: functionName,
+          }, err);
+
+          if (err.response) {
+            // The request was made and the server responded with a status code
+            // that falls out of the range of 2xx
+            errorResponse = HoldRequestConsumerError(Object.assign({}, errorResponse, {
+              message: 'received a status outside of the 2xx range from Patron Service',
+              status: err.response.status,
+              error : {
+                method: err.response.request.method,
+                url: err.response.request.path,
+                headers: err.response.headers,
+                data: err.response.data
+              }
+            }));
+          } else if (err.request) {
+            // The request was made but no response was received
+            errorResponse = HoldRequestConsumerError(Object.assign({}, errorResponse, {
+              message: 'request was made, no response from Patron Service',
+              status: 500,
+              error : err.request
+            }));
+          } else {
+            errorResponse = HoldRequestConsumerError(Object.assign({}, errorResponse, {
+              message: err.message,
+              status: err.status || 500,
+              error: err.error
+            }));
+          }
+          // Reject the promise with the proper HRC Error
+          reject(errorResponse);
+        } else {
+          const mergedArrayResults = [].concat.apply([], results);
+          resolve(mergedArrayResults);
+        }
       });
+    });
+  };
+
+  this.processBatchRequest = (itemsArray, token, type, baseApiUrl) => {
+    const functionName = 'processBatchRequest';
+    if (!token || token === '') {
```

Comment:
Instead of all these Promise.reject, does it make sense to have one if condition checking for everything that is needed and then reject just one promise. Now, this won't let you know exactly which condition failed. However, you can log the values of each parameters that you are wanting to check. The promise chain that gets the promise rejection is just logging these errors. This suggestion is to avoid all the if conditions. However, if feels necessary, please ignore my comments.